### DISCUSSION TITLE
Problem with graph.fromJSON()

### DISCUSSION THREAD

**Discussion by malomiquel** [clientIO/joint#1815] [Q&A]
Hello everyone, 
I have a problem with the function graph.fromJSON() because when i try to import a graph with a json i have an error in my console that say: "Uncaught TypeError: ModelClass is not a constructor". 
It's work with basics elements like standard.rectangl but not with my customs elements.
![example of my custom element](https://user-images.githubusercontent.com/83480447/191689230-7648a2c5-6180-4232-ae7c-57bcfba88af3.png)
<img width="323" alt="Capture d’écran 2022-09-22 094849" src="https://user-images.githubusercontent.com/83480447/191689237-8dc88192-2800-45fc-98ca-2ac48fbf4c64.png">

**Comment by jamesgeorgewilliams** [comment]
JointJS reads cell view definitions from the `joint.shapes` namespace by default. When it is looking for your custom shape, it likely cannot find it on this namespace.

You can read more about the `types` attribute in the types section in this article.
https://resources.jointjs.com/tutorial/ts-shape

Following the advice in the article above, you could assign this custom element to the `joint.shapes` namespace.

```js

const BusH = joint.dia.Element.define('BusH', {
  // attributes
});

Object.assign(joint.shapes, { BusH });

const namespace = joint.shapes;

const graph = new dia.Graph({}, { cellNamespace: namespace });
const paper = new dia.Paper({
    cellViewNamespace: namespace
});

graph.fromJSON({ 
    cells: [
        { type: 'BusH'}, 
    ]
});
```
